File: japjs-studio/src/main/java/hue/captains/singapura/japjs/studio/es/StudioCatalogue.java
package hue.captains.singapura.japjs.studio.es;

import hue.captains.singapura.japjs.core.*;
import hue.captains.singapura.japjs.studio.css.StudioStyles;

import java.util.List;

public record StudioCatalogue() implements AppModule<StudioCatalogue> {

    record appMain() implements AppModule._AppMain<StudioCatalogue> {}

    public static final StudioCatalogue INSTANCE = new StudioCatalogue();

    @Override
    public String title() {
        return "japjs · studio";
    }

    @Override
    public ImportsFor<StudioCatalogue> imports() {
        return ImportsFor.<StudioCatalogue>builder()
                .add(new ModuleImports<>(List.of(
                        new StudioStyles.st_root(),
                        new StudioStyles.st_header(),
                        new StudioStyles.st_brand(),
                        new StudioStyles.st_brand_dot(),
                        new StudioStyles.st_brand_word(),
                        new StudioStyles.st_breadcrumbs(),
                        new StudioStyles.st_crumb(),
                        new StudioStyles.st_crumb_sep(),
                        new StudioStyles.st_main(),
                        new StudioStyles.st_kicker(),
                        new StudioStyles.st_title(),
                        new StudioStyles.st_subtitle(),
                        new StudioStyles.st_section(),
                        new StudioStyles.st_section_title(),
                        new StudioStyles.st_grid(),
                        new StudioStyles.st_app_pill(),
                        new StudioStyles.st_app_pill_dark(),
                        new StudioStyles.st_app_pill_icon(),
                        new StudioStyles.st_app_pill_label(),
                        new StudioStyles.st_app_pill_desc(),
                        new StudioStyles.st_footer()
                ), StudioStyles.INSTANCE))
                .build();
    }

    @Override
    public ExportsOf<StudioCatalogue> exports() {
        return new ExportsOf<>(INSTANCE, List.of(new appMain()));
    }
}

File: japjs-studio/src/main/resources/japjs/js/hue/captains/singapura/japjs/studio/es/StudioCatalogue.js
// =============================================================================
// japjs studio — launcher / home
// Lists the available studio apps. Pre-RFC-0001, links use the legacy ?class=
// URL contract; will migrate to nav.X(...) when typed nav lands.
// =============================================================================

function appMain(rootElement) {

    function cn() {
        var parts = [];
        for (var i = 0; i < arguments.length; i++) {
            if (arguments[i]) parts.push(css.className(arguments[i]));
        }
        return parts.join(" ");
    }

    var PKG = "hue.captains.singapura.japjs.studio.es.";
    var url = function(cls) { return "/app?class=" + PKG + cls; };
    var rfcUrl = function(cls) { return "/app?class=hue.captains.singapura.japjs.studio.rfc0001." + cls; };

    var apps = [
        {
            href:     url("DocBrowser"),
            label:    "Documents",
            desc:     "Browse and read every white paper, brochure, RFC, brand artifact, and design note in the project — searchable, filterable by category, with an in-page table of contents on every doc.",
            icon:     "D",
            featured: true
        },
        {
            href:     rfcUrl("Rfc0001Plan"),
            label:    "RFC 0001 Plan",
            desc:     "Live implementation tracker for the App Registry & Typed Navigation RFC. Every step is its own URL; progress is recorded in Java code and rendered live in the studio.",
            icon:     "R",
            featured: false
        }
        // Future apps register themselves here as new tiles:
        //   { href: "...", label: "Decisions", desc: "...", icon: "L" }
        //   { href: "...", label: "Plan",      desc: "...", icon: "P" }
    ];

    var html = ''
        + '<div class="' + cn(st_root) + '">'

        // header
        + '<div class="' + cn(st_header) + '">'
        + '  <a class="' + cn(st_brand) + '" href="/app?class=hue.captains.singapura.japjs.studio.es.StudioCatalogue">'
        + '    <span class="' + cn(st_brand_dot) + '"></span>'
        + '    <span class="' + cn(st_brand_word) + '">japjs · studio</span>'
        + '  </a>'
        + '  <div class="' + cn(st_breadcrumbs) + '">'
        + '    <span class="' + cn(st_crumb) + '">Home</span>'
        + '  </div>'
        + '</div>'

        // main
        + '<div class="' + cn(st_main) + '">'
        + '  <div class="' + cn(st_kicker) + '">design &amp; project management</div>'
        + '  <h1 class="' + cn(st_title) + '">Studio</h1>'
        + '  <p class="' + cn(st_subtitle) + '">A workspace for the design, documentation, and project artifacts that drive japjs forward — built on japjs itself.</p>'

        + '  <div class="' + cn(st_section) + '">'
        + '    <div class="' + cn(st_section_title) + '">Apps</div>'
        + '    <div class="' + cn(st_grid) + '">';

    for (var i = 0; i < apps.length; i++) {
        var a = apps[i];
        var pillCls = a.featured ? cn(st_app_pill, st_app_pill_dark) : cn(st_app_pill);
        html += '<a class="' + pillCls + '" href="' + a.href + '">'
              +   '<div class="' + cn(st_app_pill_icon) + '">' + a.icon + '</div>'
              +   '<div>'
              +     '<div class="' + cn(st_app_pill_label) + '">' + a.label + '</div>'
              +     '<div class="' + cn(st_app_pill_desc) + '">' + a.desc + '</div>'
              +   '</div>'
              + '</a>';
    }

    html += '    </div>'
          + '  </div>'

          // footer
          + '  <div class="' + cn(st_footer) + '">'
          + '    Studio is a sibling Maven module to <code>japjs-demo</code>, built entirely on japjs primitives. Add new tiles to <code>StudioCatalogue.js</code>; new apps register themselves as siblings under <code>...studio.es.*</code>.'
          + '  </div>'
          + '</div>'

          + '</div>';

    rootElement.innerHTML = html;
}
